lib/graphqr/query_field.rb

```diff
@@ -56,39 +58,26 @@ def query_field(field_name, active_record_class, type_class:, scope_class: nil,
 
     private
 
-    def collection(active_record_class, type_class, scope_class)
-      Class.new(::BaseResolver) do
-        class_attribute :active_record_class
-        self.active_record_class = active_record_class
-
-        type type_class.pagination_type, null: false
-
-        argument :filter, scope_class, required: false
-
-        def resolve(filter: {})
-          authorize_graphql active_record_class, :index?
-
-          collection = apply_scopes(active_record_class, filter)
-          context[:policy_provider].authorized_records(records: collection)
-        end
-      end
+    def collection_resolver(active_record_class, type_class, scope_class)
+      resolver = base_collection_resolver(type_class, scope_class)
+      resolver.define_method(:unscoped_collection) { @unscoped_collection ||= active_record_class }
+      resolver
     end
 
-    def resource(active_record_class, type_class)
-      Class.new(::BaseResolver) do
+    def resource_resolver(active_record_class, type_class)
+      Class.new(base_resource_resolver(type_class)) do
         class_attribute :active_record_class
         self.active_record_class = active_record_class
 
-        type type_class, null: false
-
         argument :id, 'ID', required: true
 
         def resolve(id:)
-          record = self.class.active_record_class.find(id)
-
-          context[:policy_provider].allowed?(action: :show?, record: record)
+          @id = id
+          super()
```

Comment:
are () needed?

Reply:
yes,
when super doesn't accept arguments but the method calling it does, you need to specify that super is being called without args.  `super` wouldn't work here